src/exabgp/bgp/message/update/nlri/mup/t1st.py

```diff
@@ -77,23 +85,30 @@ def __eq__(self, other):
             and self.qfi == other.qfi
             and self.endpoint_ip_len == other.endpoint_ip_len
             and self.endpoint_ip == other.endpoint_ip
+            and self.source_ip_len == other.source_ip_len
+            and self.source_ip == other.source_ip
         )
 
     def __ne__(self, other):
         return not self.__eq__(other)
 
     def __str__(self):
-        return "%s:%s:%s%s:%s:%s:%s:%s" % (
+        s = "%s:%s:%s%s:%s:%s:%s%s" % (
```

Comment:
Please can you have two code path: one when source_ip_len and source_ip is not set which does not change how the object is represented, and a other branch where you can add your information, keeping the current fields unchanged and adding the two new extra ones.

What you are proposing here seems to once more change how the data would be serialised which may break code from current users.

Reply:
thank review!
I don't know if I understand the meaning well.
Is this what you're saying?
[14dd27a](https://github.com/Exa-Networks/exabgp/pull/1193/commits/14dd27a8cdce95f3626336d8408662e778782d1c)

Reply:
The output of the function should be unchanged when parsing packets which do not match the new code.

Reply:
fixed and merged